### PR TITLE
[server] - added test cases for iterator

### PR DESCRIPTION
**Pull request by LeonLiur**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-966](https://linear.app/synnaxlabs/issue/SY-966/[server]-iterator-tests)

## Description

Added test cases for iterator + fixed a bug that caused iterators on peer nodes to panic.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added sufficient regression tests to cover the changes.

## Manual QA Additions

- [x] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

### CODE REVIEW

File: synnax/pkg/distribution/framer/iterator/iterator_test.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+					Series: []telem.Series{
+						telem.NewSeriesV[int64](13, 14, 15, 16, 17),
+					}},
+				)).To(BeTrue())
+				Expect(writer.Write(core.Frame{
+					Keys: s.keys,
+					Series: []telem.Series{
+						telem.NewSeriesV[int64](18, 19, 20, 21, 22),
+					},
+				}))
```
Comment (Lham42) [inline synnax/pkg/distribution/framer/iterator/iterator_test.go:60]: Should this one have a .To(BeTrue())

Comment (LeonLiur) [reply synnax/pkg/distribution/framer/iterator/iterator_test.go:60]: Yes, you're right

Comment (LeonLiur) [reply synnax/pkg/distribution/framer/iterator/iterator_test.go:60]: Addressed

Comment (LeonLiur) [reply synnax/pkg/distribution/framer/iterator/iterator_test.go:60]: Good to merge?

**Review by Lham42 — APPROVED** [review]

**Comment by Lham42** [comment]
LGTM

**Review by emilbon99 — APPROVED** [review]

**Pull request merged**